WIP: Initial attempt at upgrade to Django 1.11

pjsier opened the pull request:
It seemed like the easiest way to research the upgrade for #326 would be to try it out and see what happens. Overall, it looks pretty doable, and this runs locally with a few caveats. I also haven't done too much testing other than clicking around and making sure templates render.

Most of the changes are relating to configuration updates and end up about the same, but there are a few that I want to flag here:

* I'm not entirely sure why, but `readonly_fields` in the admin doesn't like the `event_history_name` property method, so I removed it for now
* The API around `get_field_by_name` seems to have changed, which, if I have it implemented correctly, makes things a lot simpler
* The biggest change that I haven't updated from what I've seen so far is on `ForeignKeyRadioWidget`. It shouldn't be difficult, but Django is moving from widgets to smaller templates for form components, so we would just need to create a new template with the described behavior

There are a lot of other changes in here, and there are likely things I haven't found yet, but this seems like a way to start the conversation

cecilia-donnelly commented on streetcrm/admin.py:
```diff
@@ -723,7 +723,8 @@ class ParticipantAdmin(mixins.AdminArchiveMixin, mixins.SignInSheetAdminMixin, S
     list_display = ("name", "US_primary_phone", "institution", "participant_street_address", "participant_city_address",)
-    readonly_fields = ("action_history", "event_history_name")
+    # HAD TO REMOVE event_history_name DUE TO ISSUES
+    readonly_fields = ("action_history",)
```
In general, we try to use some searchable string across changes like this -- maybe `TODO`?

cecilia-donnelly commented on streetcrm/models.py:
```diff
@@ -136,7 +136,9 @@ class InspectMixin:
         """ Gets a field by it's name """
-        return cls._meta.get_field_by_name(name)[0]
+        # Updated https://docs.djangoproject.com/en/1.11/ref/models/meta/#retrieving-a-single-field-instance-of-a-model-by-name
+        # return cls._meta.get_field_by_name(name)[0]
```
We can remove this line, right?

cecilia-donnelly commented:
@pjsier does this need more testing from me?  What would be a helpful step for me to take?

pjsier commented:
@cecilia-donnelly if you could try it out locally and see if it runs that would be great! Other than general checks the main things I'm not sure about are `event_history_name` being removed from `readonly_fields` and the custom behavior being removed from `ForeignKeyRadioWidget`

pjsier commented:
Actually, figured out that `ForeignKeyRadioWidget` isn't being used currently (`group` was removed from tags), so that should be resolved.

cecilia-donnelly commented:
After I ran `pip install -U -r requirements.txt` and restarted the server with `python manage.py runserver`, I saw `You have 3 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): admin, auth.  Run 'python manage.py migrate' to apply them.`

Is that expected?

cecilia-donnelly commented:
They were:

```
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, sessions, streetcrm, watson
Running migrations:
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
```

cecilia-donnelly commented:
A couple issues:

- Logout fails with `'future' is not a registered tag library. Must be one of:`
- New icons for add/edit next to foreign key fields (e.g. for Organizer field on Action).  This isn't necessarily a bad thing, but we should flag that and write docs if we're going to keep them.
- New helptext "type some text to search in this autocomplete" is surprising

pjsier commented:
@cecilia-donnelly ah yeah, I forgot to mention the migrations. Is there anywhere in the documentation I could flag that (it would only really be for the upgrade)? If it makes sense, I'll add a section to the bottom of `INSTALL.md` similar to the for `Old Versions`. On the other points:

* The `future` sounds straightforward. I removed it somewhere else, so looks like I missed that one
* Good to know on the icons, and I'll add it to the docs unless there's a reason for not changing them
* The helptext is definitely weird and I can change that back

I can also go back through the history to figure out what to do with `event_history_name`, and I'm guessing that'll be straightforward. Thanks for the feedback on this!

pjsier commented:
The `future` tag was actually from `django-admin-bootstrapped` which doesn't seem to have been maintained. [This fork](https://github.com/ZeroCater/django-admin-bootstrapped) made some pretty basic changes and is working, so I added that specific commit to the `requirements.txt`.

I also added some initial docs on the upgrade to `INSTALL.md`, but I'll have to dig a bit more to find where the helptext is coming from

pjsier commented:
So looking more into the placeholders for autocomplete fields, and it seems like that's a small bug that pointed out a bigger issue. [django-autocomplete-light](https://django-autocomplete-light.readthedocs.io/en/master/#upgrading) version 2 apparently doesn't support above Django 1.9, and it sounds like the upgrade to 3 isn't backwards-compatible. Let me know how you want to move forward with that

cecilia-donnelly commented:
@pjsier I see.  Thanks for the update, and for your patience!  In that case I think we should get #325 merged as-is (I'll re-review right now).  We can file this as tech debt for now -- we might come back to it quickly but it doesn't sound like we need it right away.

So I'm clear: upgrading to django-autocomplete-light v3 would mean changing the way we're using it, right?  Did you look into that at all, or just enough to see that it would be a breaking change?

pjsier commented:
@cecilia-donnelly sounds good! And yeah, it sounds like there are breaking changes and significantly different ways of doing things. I started taking a look at the details, but it started to seem like it could take some time so I figured I would check in first

cecilia-donnelly commented:
@pjsier are you okay with closing this for now?  We can always reopen it later when we're ready to pick this back up.

pjsier commented:
@cecilia-donnelly sounds good to me!
